sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/orc/OrcPartitionReaderFactory.scala

```diff
@@ -158,6 +169,10 @@ case class OrcPartitionReaderFactory(
       val taskAttemptContext = new TaskAttemptContextImpl(taskConf, attemptId)
 
       val batchReader = new OrcColumnarBatchReader(capacity)
+      if (orcMetaCacheEnabled) {
+        val tail = OrcFileMeta.readTailFromCache(filePath, conf)
```

Comment:
If this calls the following, `OrcFileMetaKey.apply` always seems to read files via `Utils.tryWithResource(new ReaderImpl(path, readerOptions)`. Do I understand correctly? 
```
  def readTailFromCache(path: Path, conf: Configuration): OrcTail =
     readTailFromCache(OrcFileMetaKey(path, conf))
```

Reply:
No, 
```
def readTailFromCache(path: Path, conf: Configuration): OrcTail =
     readTailFromCache(OrcFileMetaKey(path, conf))
```

equals 

```
def readTailFromCache(path: Path, conf: Configuration): OrcTail =
     FileMetaCacheManager.get(OrcFileMetaKey(path, conf)).asInstanceOf[OrcFileMeta].tail
```

And `OrcFileMeta.apply` will trigger file read but `OrcFileMetaKey.apply`  does not trigger file read

Reply:
42d2bfd change 

```
def readTailFromCache(path: Path, conf: Configuration): OrcTail =
     readTailFromCache(OrcFileMetaKey(path, conf))
```
to

```
def readTailFromCache(path: Path, conf: Configuration): OrcTail =
     FileMetaCacheManager.get(OrcFileMetaKey(path, conf)).asInstanceOf[OrcFileMeta].tail
```

and remove a redundant method

Reply:
Got it. Right. I was confused with `OrcFileMetaKey()` and `OrcFileMeta`.

Reply:
Are there any other better naming suggestions?

Reply:
For now, it looks okay to me.